src/base-app/event.cpp

```diff
@@ -0,0 +1,99 @@
+/*Implemented by Jake*/
+
+#include "event.h"
+
+namespace base{
+    event::event()
+    {
+        mainHost = NULL;
```

Comment:
The use of NULL is a little confusing. Typically you only use NULL when working with pointers.

Reply:
Writing: `mainHost = ""` would be more clear. Technically this is unnecessary, the default constructor will give it the same value.